prow/pod-utils/clone/clone.go

```diff
@@ -261,7 +264,7 @@ func (g *gitCtx) commandsForBaseRef(refs prowapi.Refs, gitUserName, gitUserEmail
 	}
 
 	{
-		fetchArgs := append([]string{}, depthArgs...)
+		fetchArgs := append([]string{}, fetchOpts...)
```

Comment:
Aside: The existing code here is a bit confusing... I don't understand why this needs to be in a separately delimited block, and why we are creating fetchArgs anew inside here. There's definitely room for cleanup here in the future.

Reply:
I've used the block to create a new fetchArgs var, so it's useful and hopefully clear to the reader that it controls the scope of the variable.

Reply:
And to clarify why it is built again, the previous fetchArgs already contained the URI and some options not used by the fetch created here.